### ISSUE TITLE
Billboard,Label . When used together, the label cannot be displayed at the top layer when overlapped

### ISSUE THREAD

**Issue by zhangdaoke** [CesiumGS/cesium#11496]
![image](https://github.com/CesiumGS/cesium/assets/135095609/32de5959-fd45-47a6-b2c6-f460b846ffe7)

Partial code:

 function addPrimitiveBillboard(id: string, pointCoord: Cesium.Cartesian3, billboardCollection: Cesium.BillboardCollection) {
        billboardCollection.add({
            id,
            position: pointCoord,
            image: circl_png,
            horizontalOrigin: Cesium.HorizontalOrigin.CENTER, // default
            verticalOrigin: Cesium.VerticalOrigin.CENTER, // default: CENTER
            scale: 1,
            width: 53, // default: undefined
            height: 53, // default: undefined
            eyeOffset: new Cesium.Cartesian3(0, 0, 10),
            disableDepthTestDistance: Number.POSITIVE_INFINITY,
            distanceDisplayCondition: new Cesium.DistanceDisplayCondition(0, 100000),
        });
    }

function  addPrimitiveLabel(id: string, pointCoord: Cesium.Cartesian3, text, lableCollection: Cesium.LabelCollection) {
        lableCollection.add({
            id,
            position: pointCoord,
            pixelOffset: new Cesium.Cartesian2(-20, 10),  //60 -37
            show: true, // default
            text,
            scale: 0.45,
            font: "800 35px monospace",// 15pt monospace
            style: Cesium.LabelStyle.FILL_AND_OUTLINE,
            showBackground: false,
            backgroundColor: new Cesium.Color(0.621, 0.46, 0.11, 0.1),
            fillColor: Cesium.Color.WHITE,
            horizontalOrigin: Cesium.HorizontalOrigin.LEFT,
            verticalOrigin: Cesium.VerticalOrigin.BOTTOM,
            eyeOffset: new Cesium.Cartesian3(0, 0, -0.5),
            disableDepthTestDistance: Number.POSITIVE_INFINITY,
            distanceDisplayCondition: new Cesium.DistanceDisplayCondition(0, 100000),
        })
    }

**Comment by zhangdaoke** [comment]
sandcastle:
https://sandcastle.cesium.com/#c=1VZhb+I4EP0rFtKxqRSchBRKOehdC3SLRGF1zXW1ElLlJA741olztlNgV/3vN0npbhLoaq+6L4cQcmb8nmc88yYEIlEaPTK6oRINUUI3aEQVy2J8X9iMZSMonkci0YQlVC4bJ78uE0414sSnfCQ4p4FmIgH4Mw9WAU0oTiWLmWaPVGEShkaJelYFGl9RgehX8Ojp5OUgn3HuCyLDtxx2dQj+4YHLBAg+vNB9QxvvHMd5Z+5JR0RqWJHExZEU8ZiuJKXKOLdN1z0xj8R7UqUtbuDfUC4ba8q5QBshebhsmLW7B/p9KgGJqSQ44jtPGF+XCSp9Qqo0S0gO6KOfO7aKz+Qe7ILnCQ6NsuS5HMfvjIUmSgVL9EgICetjF4NKQR7xF+Ws5QG0VUMqFHsOrHRaDROTFe3DNWqdqr5lFc8KZ4lKOVFrHIjYStdCi5bTbbunjn3qnrtdp9Vuk8A5bzvdztnZb2zLmT+UfusU29htsi0Lh7fuZnv716Ubja96tzfbDXl/bvs7Z/PpY8ijm20ven+f5b5iffmLO4Zvk2RaDCMhY6KbEdPDQIq0uRm2nd5Z8+9hz4YKV8NfC8m+5BLkC8lW7Hv9bmoOPJrMvckfJrIsqHdEMq6rTI9UahYc8NxXzEdYYGdhq7IpgICOnFq4GxbqdR913CpDlsAShkhYS46y1Vr/9Ha6o4soUhR2lHT+vZEN20TwdeyD/mWK+JyOaarXHmhhzJQmSQDxz7PYB/F8WNxNven95GE6v57Op96nQ4ICAEDomh3MxHDfd6VAxq9sMoqY8k85rqd85hSLfPS8oqjncVFXk6ZbbeZTmNNX9FTz/VdaStmW8h+VoG202vsKIKho10Yt96zWOGux6SMtM/p6qxYJHu03G592ap4IRNBHy0bPtpHbSbcoFolQKQkoiAlOcDqpLtmqrHqXs5ZfT3e5CV9PZ7OHy/n4YfGnN5vOJ+ZhElck+LySAnq1jyLCFa3t8b/5oRBCVi8stxg27rYdM8+pm/86xdqpN28Es3HPUEbjjzdTb/LmaTGbXHvmWwbE1cLzFrfmm5TZsnHn/6HNvTQbZmNQ9MjFi/N3FqdCapRJbmBsaRoDH2Rp+VnwmWocKPUi7IFVhg5C9giiGx75X4UCeBMp8EQZ53fsCzTuxcCC/QdQLkjIktUCSgVZ5NvWzsXs2YgxHljweBypheA+kTXmfwA

**Comment by ggetz** [comment]
Hi @zhangdaoke, in your code example, I am seeing the following result regardless of the z-values for eyeOffset. Is that also what your are seeing?

<img width="183" alt="image" src="https://github.com/CesiumGS/cesium/assets/4439461/16e18c51-2444-49b3-a939-20249cc054a7">

It appears like some sort of blending problem.

**Comment by zhangdaoke** [comment]
@ggetz Yes, I just want the text to be very clear and visible when overlapped with the image, but no matter what I try, it doesn't work. I don't really want to achieve this by creating elements in HTML because it would be very difficult to manage.

**Comment by zhangdaoke** [comment]
@ggetz 
![image](https://github.com/CesiumGS/cesium/assets/135095609/22ff6fc6-6f5c-4a11-a7f1-c94afbd55580)
I want to use the background box on the right as the background for the text, instead of the simpler one on the left.

**Comment by ggetz** [comment]
> I want to use the background box on the right as the background for the text, instead of the simpler one on the left.

@zhangdaoke For that, I would suggest using an [HTML overlay](https://sandcastle.cesium.com/?src=HTML%20Overlays.html). It allow you to use full HTML and CSS styling to render labels.

**Comment by zhangdaoke** [comment]
@ggetz Okay, because I believe that when the camera moves, the position of the billboard may not be recalculated; it's just that the camera's position has changed. This default approach should provide better performance. Therefore, I'm very concerned that implementing this process using HTML (where there are tens of thousands of such POIs in the actual scenario) in the manner shown in the image below will result in a significant performance loss.
![image](https://github.com/CesiumGS/cesium/assets/135095609/b2c4e6bc-df85-4060-8afa-838954209cea)

**Comment by onsummer** [comment]
It seems that label and billboard have the same pass order?

**Comment by ggetz** [comment]
Maybe related to https://github.com/CesiumGS/cesium/issues/9681

**Comment by zhangdaoke** [comment]
@ggetz  https://github.com/CesiumGS/cesium/issues/11496#issuecomment-1706574241,The html implementation of billboard may not be able to do this. I am currently trying to use html to generate billboard. However, this approach presents a significant and insurmountable issue: the 'HTML-Billboard' requires click events, which unfortunately obstruct Cesium events. When the number of "HTML-Billboards" on the page reaches hundreds, it becomes highly probable that the user's mouse position will be situated on the 'html-billboard', thereby impacting essential mouse and wheel events in Cesium, including rotation, rendering them ineffective. Although we have actively attempted to resolve this problem, no satisfactory solution has been attained thus far. Hence, I am documenting and refuting your proposed solution for your reference.

**Comment by syzdev** [comment]
Using `canvas` to create `Billboard` might be a good solution to this problem.

**Comment by syzdev** [comment]
@zhangdaoke 

If the content to be displayed is so complex, you can only use `HTML overlay`, but as you said, you have to make difficult decisions in the mouse events of `HTML DOM Element` and `Scene.camera`.
